### PR TITLE
Fix energy conservation

### PR DESCRIPTION
**Pull request by unoebauer**
As found by @chvogl and reported in issue #448, an inconsistency in the energy conservation calculation exists. Instead of the Doppler factor at the interaction location, the initial Doppler factor, determined at the beginning of the trajectory segment is used. For typical applications, the consequences of this inconsistency should be small (as discussed in #448). Nevertheless, this PR aims at fixing this inconsistency.

- [x] fix  energy conservation, i.e. use Doppler factors at the interaction location
- [x] investigate consequences (check tardis_example)
- [x] decide whether move_packet should actually return the doppler factor
- [x] adjust and update tests

### CODE REVIEW

**Comment by yeganer** [comment]
@unoebauer When I first looked at the C code I was wondering why move_packed returned the doppler_factor.

My suggestion: add `rpacket_get_doppler_factor(*rpacket) `returning the dopplerfactor of the current packet ( current r, mu). I don't think we need that factor saved to rpacket. then any time we need the doppler factor we can just call that function and save it in a local variable.
If there are too many calls to `rpacket_get_doppler_factor`, maybe we should rename it to `update_doppler_factor` and then save the doppler_factor as a property of rpacket

**Comment by unoebauer** [comment]
Fix does not seem to induce significant changes for standard setups:

spec_cmp

**Comment by unoebauer** [comment]
@yeganer, @ssim, @wkerzendorf, @chvogl 

Many tests had to be updated - for future references: we should write a script which recalculates all the data files, once critical changes have been made within the Monte Carlo routine. Of course, before using these files, they should be carefully compared with the current ones. I have started a very very crude version of such a script.

**Comment by yeganer** [comment]
@unoebauer Last time we did this I advised against creating such a script because it could be used carelessly. But I guess with future slow tests coming we might need that.

**Comment by unoebauer** [comment]
Well - true, there is a temptation to use it carelessly. But manually building all the data files is really painfull!

**Comment by unoebauer** [comment]
Ok - Travis is finally happy. Should this be merged @wkerzendorf, @yeganer?

It would be good if you, @ssim and @chvogl, could also have a careful look at it.

File: tardis/montecarlo/src/test_cmontecarlo.c [hunk 1]
```diff
@@ -261,17 +261,17 @@ test_rpacket_doppler_factor(void){
...
-	double DISTANCE = 1e13;
-        rpacket_t rp;
-        storage_model_t sm;
-        init_rpacket(&rp);
-        init_storage_model(&sm);
-	double res = move_packet(&rp, &sm, DISTANCE);
-        dealloc_storage_model(&sm);
-        return res;
-}
+//double
```
Comment (wkerzendorf) [inline tardis/montecarlo/src/test_cmontecarlo.c]: @unoebauer should we just delete these lines?

Comment (unoebauer) [inline tardis/montecarlo/src/test_cmontecarlo.c]: Well - I wouldn't for now. Eventually, we want to test move_packet, just in a different way. I'd keep these lines as a reminder.

**Comment by yeganer** [comment]
I think this is ready to get merged.

Regarding `test_move_packet`:
I think it's fine to have it in the c code. We could even uncomment it because the test is skipped anyway. Deleting the lines is dangerous because when the tests get an overhaul one could easily forget implementing this one.

**Comment by ssim** [comment]
I think that yes, I agree with this fix. I wonder how it came about that there was an error here but I think that yes, it is was wrong. It would have been correct if the Dopper factor were re-calculated at the end of "move packet" before being returned, but that was not being done. This seems like the right fix to me.

**Comment by chvogl** [comment]
I agree that this can and should be merged.

**Pull request merged**